pvlib/windspeed.py

```diff
@@ -0,0 +1,165 @@
+
+    exponent : float, optional
+        Exponent based on the surface type. [-]
+
+    surface_type : string, optional
+        If supplied, overrides ``exponent``. Can be one of the following
+        (see [1]_):
+
+        * ``'unstable_air_above_open_water_surface'``
+        * ``'neutral_air_above_open_water_surface'``
+        * ``'stable_air_above_open_water_surface'``
+        * ``'unstable_air_above_flat_open_coast'``
+        * ``'neutral_air_above_flat_open_coast'``
+        * ``'stable_air_above_flat_open_coast'``
+        * ``'unstable_air_above_human_inhabited_areas'``
+        * ``'neutral_air_above_human_inhabited_areas'``
+        * ``'stable_air_above_human_inhabited_areas'``
+
+    Returns
+    -------
+    wind_speed : numeric
+        Adjusted wind speed for the desired height. [m/s]
+
+    Raises
+    ------
+    ValueError
+        If neither of ``exponent`` nor a ``surface_type`` is given.
+        If both ``exponent`` nor a ``surface_type`` is given. These parameters
+        are mutually exclusive.
+
+    KeyError
+        If the specified ``surface_type`` is invalid.
+
+    Notes
+    -----
+    The equation for calculating the wind speed at a height of :math:`h` is
+    given by the following power law equation [1]_, [2]_:
+
+    .. math::
+       :label: wind speed
+
+        windspeed_h = windspeed_{href} \cdot \left( \frac{h}{h_{ref}} \right)^a
+
+    where :math:`h` [m] is the height at which we would like to calculate the
+    wind speed, :math:`h_{ref}` [m] is the reference height at which the wind
+    speed is known, and :math:`windspeed_h` [m/s] and :math:`windspeed_{href}`
+    [m/s] are the corresponding wind speeds at these heights. :math:`a` is a
+    value that depends on the surface type. Some values found in the literature
+    [1]_ for :math:`a` are the following:
+
+    .. table:: Values for the Hellmann-exponent
+
+    +-----------+--------------------+------------------+------------------+
+    | Stability | Open water surface | Flat, open coast | Cities, villages |
+    +===========+====================+==================+==================+
+    | Unstable  | 0.06               | 0.10             | 0.27             |
+    +-----------+--------------------+------------------+------------------+
+    | Neutral   | 0.11               | 0.16             | 0.40             |
+    +-----------+--------------------+------------------+------------------+
+    | Stable    | 0.27               | 0.34             | 0.60             |
+    +-----------+--------------------+------------------+------------------+
+
+    In a report by SANDIA [3]_, this equation was experimentally tested for a
+    height of 30 ft (9.144 m) and the following coefficients were recommended:
+    :math:`h_{ref} = 9.144` [m], :math:`a = 0.219` [-], and
+    :math:`windspeed_{href}` is the wind speed at a height of 9.144 [m].
+
+    It should be noted that the equation retuns a value of NaN if the
```

Comment:
I don't know if it is better, or makes no difference, to write ``np.nan`` rather than NaN in this situation. Corrected a minor typo in this line anyway though.
```suggestion
    It should be noted that the equation returns a value of NaN if the
```